src/dev/i18n/extract_i18n_call_messages.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  isObjectExpression,
+  isStringLiteral,
+  isTemplateLiteral,
+} from '@babel/types';
+
+import { isPropertyWithKey, escapeLineBreak } from './utils';
+import { DEFAULT_MESSAGE_KEY } from './constants';
+
+/**
+ * Extract messages from `funcName('id', { defaultMessage: 'Message text' })` call expression AST
+ */
+export function extractI18nCallMessages(node) {
+  const [idSubTree, optionsSubTree] = node.arguments;
+
+  if (!isStringLiteral(idSubTree)) {
```

Comment:
__nit:__ since we have `if (!isStringLiteral(idSubTree)) {` with `throw` already, it'd make sense to add one more right after that and have less nested code:
```js
if (!isObjectExpression(optionsSubTree)) {
    throw new Error(
      `Object with defaultMessage property is not provided for ${messageId} id.`
    );
  }
```